src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2517,6 +2519,11 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                     privateKeyOpt = StrongNameKeys.PrivateKey;
                 }
 
+                if (!options.EmitMetadataOnly && CommonCompiler.HasUnsuppressedErrors(diagnostics))
```

Comment:
>CommonCompiler.HasUnsuppressedErrors(diagnostics) [](start = 49, length = 49)

Have we applied pragma suppression and other diagnostic filtering by this point? Where did we do this?

Reply:
I believe diagnostic filtering is applied towards the end of each of the compilation stages. For example, applied [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/CSharpCompilation.cs,2689) for `CompileMethods`, applied [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/CSharpCompilation.cs,2732) for `GenerateResourcesAndDocumentationComments` (also verified identical implementation in VB compilation)